shared-module/vectorio/VectorShape.c

```diff
@@ -224,20 +228,43 @@ void common_hal_vectorio_vector_shape_set_location(vectorio_vector_shape_t *self
     mp_obj_t *tuple_items;
     mp_obj_tuple_get(xy, &tuple_len, &tuple_items);
     if (tuple_len != 2) {
-        mp_raise_TypeError_varg(translate("(x,y) integers required"));
+        mp_raise_TypeError(translate("(x,y) integers required"));
     }
 
     mp_int_t x;
     mp_int_t y;
     if (!mp_obj_get_int_maybe(tuple_items[ 0 ], &x)
-        || !mp_obj_get_int_maybe(tuple_items[ 1 ], &y)
-        || x < SHRT_MIN || x > SHRT_MAX || y < SHRT_MIN || y > SHRT_MAX
-        ) {
+        || !mp_obj_get_int_maybe(tuple_items[ 1 ], &y)) {
         mp_raise_ValueError_varg(translate("unsupported %q type"), MP_QSTR_point);
     }
-    self->x = (int16_t)x;
-    self->y = (int16_t)y;
-    common_hal_vectorio_vector_shape_set_dirty(self);
+    bool dirty = false;
+    if (self->x != x) {
+        vectorio_vector_shape_validate_x_bounds(x);
+        self->x = x;
+        dirty = true;
+    }
+    if (self->y != y) {
+        vectorio_vector_shape_validate_y_bounds(y);
+        self->y = y;
+        dirty = true;
+    }
+    if (dirty) {
+        common_hal_vectorio_vector_shape_set_dirty(self);
+    }
+}
+
+
+void vectorio_vector_shape_validate_x_bounds(mp_int_t x) {
```

Comment:
I think these methods should be static (and therefore not in headers) right?

Also notice that everywhere self->y is set you also call this. Maybe it should be `static void set_y(vectorio_vector_shape_t *self, mp_int_t y)` ? Then just replace every self->y = y with set_y(self, y) and this looks right otherwise